src/core/core.datasetController.js

```diff
@@ -389,9 +390,13 @@ helpers.extend(DatasetController.prototype, {
 	 */
 	_resolveDataElementOptions: function(element, index) {
 		var me = this;
+		var custom = element && element.custom;
+		var cached = me._cachedDataOpts;
+		if (cached && !custom) {
+			return helpers.extend({}, cached);
```

Comment:
This is safer, but I also worry that copying the options for every data point could be expensive for large charts. What if we instead did something like this below when setting me._cachedDataOpts:

```
		if (info.cacheable) {
			me._cachedDataOpts = Object.freeze(helpers.extend({}, cached));
		}
```

Reply:
When its not frozen, `bubble` can use the caching too and just resolve the `radius` part always (could also cache that, but I didn't bother).
I agree that cloning is not optimal. But some parts modify the options directly (some bubble tests fail). I'll try to fix those instead.